### DISCUSSION TITLE
Using FB Passthrough Ext

### DISCUSSION THREAD

**Discussion by jackdaus** [StereoKit/StereoKit#377] [Q&A] [answered]
I saw that a tool for the FB Passthrough Ext was released a few months ago (https://github.com/StereoKit/StereoKit/blob/develop/Examples/StereoKitTest/Tools/PassthroughFBExt.cs). I'm trying to enable the passthrough feature in my solution, but I can't quite get it to work. When I try to add the PassthroughFBExt IStepper, I get an error message:

> [StereoKit] [SK error] backend_openxr_ functions only work when OpenXR is the backend!

I logged the Backend.XRType, and it looks like it's == Simulator. 

Here is some info about the project that may be helpful:
StereoKit version: 0.3.6-preview.6
Device: Oculus Quest 2
The solution is based on the StereoKitXPlat repo. You can find my code here: https://github.com/jackdaus/StereoKitXPlat-PaintTutorial-Port

So it looks like the the backend needs to be OpenXR in order to use the FB Passthrough Ext. But shouldn't my backend be OpenXR if I'm running it on my Quest 2? Is there a way I can manually change the backend so that it is using OpenXR?

**Comment by maluoi** [comment, marked as answer]
Hmm, it could be a misleading error message in this particular case? I do see you calling `passthroughStepper.EnabledPassthrough = true;` immediately after adding it as a stepper, but before StereoKit has been initialized! I'd hazard a guess that it's thinking the backend is Simulator because _no_ backend has been initialized at that point :D

Try calling `passthroughStepper.EnabledPassthrough = true;` _after_ `SK.Initialize`, I think that'll do it for you!

And in the meantime, I'll look into making that class a little more bulletproof. I think that's a scenario it should probably handle much more gracefully!

**Comment by jackdaus** [comment]
Thanks, that fixed it! I moved `passthroughStepper.EnabledPassthrough = true;` to after the SK.Initialize call. So now I have something like


```
                var passthroughStepper = SK.AddStepper(new PassthroughFBExt());

                if (!SK.Initialize(settings))
			return;	
		app.Init();

		passthroughStepper.EnabledPassthrough = true;
```

Thanks again!
